src/asInput/index.jsx

```diff
@@ -149,21 +163,28 @@ const asInput = (WrappedComponent, labelFirst = true) => {
     render() {
       const { description, error, describedBy } = this.getDescriptions();
       return (
-        <div className={styles['form-group']}>
-          {labelFirst && <label id={`label-${this.state.id}`} htmlFor={this.state.id}>{this.props.label}</label>}
+        <div className={[classNames({
+            [styles['form-group']]: inputType !== 'checkbox',
+            [styles['form-check']]: inputType === 'checkbox',
+          })]}
+        >
+          {labelFirst && this.getLabel()}
           <WrappedComponent
             {...this.props}
             {...this.state}
             className={[classNames(
-              styles['form-control'],
-              { [styles['is-invalid']]: !this.state.isValid && this.hasDangerTheme() },
+              {
+                [styles['form-control']]: inputType !== 'checkbox',
+                [styles['form-check-input']]: inputType === 'checkbox',
+                [styles['is-invalid']]: !this.state.isValid && this.hasDangerTheme(),
+              },
```

Comment:
Bootstrap has a different class name to be applied to the input field when that input is a checkbox. https://getbootstrap.com/docs/4.0/components/forms/#checkboxes-and-radios